dsl_parser/tasks.py

```diff
@@ -144,12 +144,13 @@ def _validate_secrets(plan, get_secret_method):
         )
 
 
-def prepare_deployment_plan(plan, get_secret_method=None, inputs=None, **_):
+def prepare_deployment_plan(plan, get_secret_method=None, inputs=None,
+                            functions_eager=True, **_):
     """
     Prepare a plan for deployment
     """
     plan = models.Plan(copy.deepcopy(plan))
     _set_plan_inputs(plan, inputs)
-    _process_functions(plan)
+    _process_functions(plan, functions_eager)
```

Comment:
if we will stick with this flag, I think maybe to switch this to eager_evaluation